pkg/config/config.go

```diff
@@ -313,14 +318,36 @@ func (dc *DeploymentConfig) ExpandConfig() error {
 	return nil
 }
 
+func (dc *DeploymentConfig) addModuleConnection(ref reference, kind connectionKind, sharedVariables []string) error {
+	if dc.moduleConnections == nil {
+		dc.moduleConnections = make(map[string][]ModConnection)
+	}
+
+	if !kind.IsValid() {
+		return fmt.Errorf(unexpectedConnectionKind)
```

Comment:
nit. I think it's ok to `fatal` here and don't bother with propagating error. `kind.IsValid()` is something unlikely to happen in the wild and clearly caused by dev-error and most likely will be caught during development.